pkg/cmd/kafka/consumergroup/resetoffset/reset_offset.go

```diff
@@ -0,0 +1,209 @@
+	"github.com/redhat-developer/app-services-cli/pkg/cmdutil"
+	flagutil "github.com/redhat-developer/app-services-cli/pkg/cmdutil/flags"
+	"github.com/redhat-developer/app-services-cli/pkg/connection"
+	"github.com/redhat-developer/app-services-cli/pkg/iostreams"
+	"github.com/redhat-developer/app-services-cli/pkg/localize"
+	"github.com/redhat-developer/app-services-cli/pkg/logging"
+	kafkainstanceclient "github.com/redhat-developer/app-services-sdk-go/kafkainstance/apiv1internal/client"
+	"github.com/spf13/cobra"
+)
+
+type Options struct {
+	kafkaID     string
+	id          string
+	skipConfirm bool
+	value       string
+	offset      string
+	topic       string
+	partitions  string
+
+	IO         *iostreams.IOStreams
+	Config     config.IConfig
+	Connection factory.ConnectionFunc
+	Logger     func() (logging.Logger, error)
+	localizer  localize.Localizer
+}
+
+// NewResetOffsetConsumerGroupCommand gets a new command for resetting offset for a consumer group.
+func NewResetOffsetConsumerGroupCommand(f *factory.Factory) *cobra.Command {
+	opts := &Options{
+		Connection: f.Connection,
+		Config:     f.Config,
+		IO:         f.IOStreams,
+		Logger:     f.Logger,
+		localizer:  f.Localizer,
+	}
+
+	cmd := &cobra.Command{
+		Use:     "reset-offset",
+		Short:   opts.localizer.MustLocalize("kafka.consumerGroup.resetOffset.cmd.shortDescription"),
+		Long:    opts.localizer.MustLocalize("kafka.consumerGroup.resetOffset.cmd.longDescription"),
+		Example: opts.localizer.MustLocalize("kafka.consumerGroup.resetOffset.cmd.example"),
+		Args:    cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, args []string) (err error) {
+
+			if opts.offset != "" {
+				if err = flag.ValidateOffset(opts.offset); err != nil {
+					return err
+				}
+			}
+
+			if opts.kafkaID != "" {
+				return runCmd(opts)
+			}
+
+			cfg, err := opts.Config.Load()
+			if err != nil {
+				return err
+			}
+
+			if !cfg.HasKafka() {
+				return errors.New(opts.localizer.MustLocalize("kafka.consumerGroup.common.error.noKafkaSelected"))
+			}
+
+			opts.kafkaID = cfg.Services.Kafka.ClusterID
+
+			return runCmd(opts)
+		},
+	}
+
+	cmd.Flags().BoolVarP(&opts.skipConfirm, "yes", "y", false, opts.localizer.MustLocalize("kafka.consumerGroup.resetOffset.flag.yes"))
+	cmd.Flags().StringVar(&opts.id, "id", "", opts.localizer.MustLocalize("kafka.consumerGroup.common.flag.id.description", localize.NewEntry("Action", "reset-offset")))
+	cmd.Flags().StringVar(&opts.value, "value", "", opts.localizer.MustLocalize("kafka.consumerGroup.resetOffset.flag.value"))
+	cmd.Flags().StringVar(&opts.offset, "offset", "", opts.localizer.MustLocalize("kafka.consumerGroup.resetOffset.flag.offset"))
+	cmd.Flags().StringVar(&opts.topic, "topic", "", opts.localizer.MustLocalize("kafka.consumerGroup.resetOffset.flag.topic"))
+	cmd.Flags().StringVar(&opts.partitions, "partitions", "", opts.localizer.MustLocalize("kafka.consumerGroup.resetOffset.flag.partitions"))
```

Comment:
for cases where a flag accepts multiple individual values, use `Int32SliceVarP`:

```shell
// Int32SliceVarP is like Int32SliceVar, but accepts a shorthand letter that can be used after a single dash.
func (f *FlagSet) Int32SliceVarP(p *[]int32, name, shorthand string, value []int32, usage
}
```

Reply:
Had no idea cobra supports these type of parameters. Thanks!